#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "pid"

gen = ParameterGenerator()

gen.add(
    "pid_p",
    double_t,
    0,
    "Specifies the p of the PID-Controller.",
    4.0,
    0.0001,
    10
)

gen.add(
    "pid_i",
    double_t,
    0,
    "Specifies the i of the PID-Controller.",
    0.2,
    0.0001,
    10
)

gen.add(
    "pid_d",
    double_t,
    0,
    "Specifies the d of the PID-Controller.",
    0.02,
    0.0001,
    10
)

gen.add(
    "anti_windup",
    double_t,
    0,
    "Specifies the anti_windup of the PID-Controller.",
    0.2,
    0.0001,
    10
)

exit(gen.generate(PACKAGE, "pid", "pid"))